source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -0,0 +1,223 @@
+#include "AMDSClientTCPSocket.h"
+
+#include "source/AMDSDataStream.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+#include "source/ClientRequest/AMDSClientRequestSupport.h"
+#include "source/ClientRequest/AMDSClientRequest.h"
+#include "source/ClientRequest/AMDSClientIntrospectionRequest.h"
+#include "source/ClientRequest/AMDSClientStatisticsRequest.h"
+#include "source/ClientRequest/AMDSClientStartTimePlusCountDataRequest.h"
+#include "source/ClientRequest/AMDSClientRelativeCountPlusCountDataRequest.h"
+#include "source/ClientRequest/AMDSClientStartTimeToEndTimeDataRequest.h"
+#include "source/ClientRequest/AMDSClientMiddleTimePlusCountBeforeAndAfterDataRequest.h"
+#include "source/ClientRequest/AMDSClientContinuousDataRequest.h"
+#include "source/util/AMDSErrorMonitor.h"
+
+AMDSClientTCPSocket::AMDSClientTCPSocket(const QString host, const quint16 port, QObject *parent)
+	:QObject(parent)
+{
+	socketKey_ = "";
+	tcpSocket_ = new QTcpSocket(this);
+
+	connect(tcpSocket_, SIGNAL(readyRead()), this, SLOT(readFortune()));
+	connect(tcpSocket_, SIGNAL(error(QAbstractSocket::SocketError)), this, SLOT(onSocketError(QAbstractSocket::SocketError)));
+
+	tcpSocket_->connectToHost(host, port);
+}
+
+AMDSClientTCPSocket::~AMDSClientTCPSocket()
+{
+	tcpSocket_->abort();
```

Comment:
Would it make more sense to use `disconnectFromHost()` here?  Is it mandatory that the socket be closed abruptly rather than smoothly?